dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/SdkFeature.kt

```diff
@@ -156,6 +159,36 @@ internal class SdkFeature(
         )
     }
 
+    override fun <T : Any> writeToDataStore(
+        dataStoreFileName: String,
+        featureName: String,
+        serializer: Serializer<T>,
+        data: T
+    ) {
+        coreFeature.dataStoreHandler.write(
+            dataStoreFileName = dataStoreFileName,
+            featureName = featureName,
+            serializer = serializer,
+            data = data
+        )
+    }
```

Comment:
it means the caller should care about calling it on the worker thread? should we do an explicit dispatch to the persistence thread here? don't know the usage patterns though

Reply:
I agree with @0xnm here. I would rather use the persistence thread here and make this operations serial. This will also get rid of all the possible race conditions you could experience.